### ISSUE TITLE
fuzzy matcher to find files with representative letters

### ISSUE THREAD

**Issue by keitee** [Shougo/unite.vim#868]
Hello,

Not sure if unite already supports this and do not use the issue template since it's more of feature question. Please see this example.

Suppose that there is a file, SampleComponentEvent.h/cpp, and runs unite search with fuzzy matcher. Types "SCE" in input:

1) unite do not show up this match but need to type one by one to narrow down.
2) ctrlp do shows up this match.

Given that there are a lot of files which have such long names in cpp projects, would be great to have it in unite as well. So wonders if it's something to have by changing configuration in unite or to be a feature to enhance. Please advise.

Many thanks for your great work.

**Comment by Shougo** [comment]
It is ranking problem.  Yes, "sorter_rank" does not recognize the letters.
But I don't want to implement the feature.  Pull request is wellcome.

**Comment by atomictom** [comment]
It looks like 'sorter_selecta' is close, but it only supports acronyms on word boundaries, not case boundaries. Maybe you could ask about it [in that project](https://github.com/garybernhardt/selecta)?

**Comment by Shougo** [comment]
Unfortunatelly, I don't implement "sorter_selecta".
I don't want to change the behavior.

https://github.com/Shougo/unite.vim/pull/544

**Comment by atomictom** [comment]
Oh, I meant that keitee could ask in that project =). I appreciate what you've done. 

But I see that the implementation is actually just copied from the selecta project. So any changes would need to be a pull request for unite.

**Comment by Shougo** [comment]
Yes.

**Comment by JCavallo** [comment]
I included support for CamelCase filtering in my tentative to port sorter_selecta to python :+1: 
https://github.com/Shougo/unite.vim/pull/1004

**Comment by Shougo** [comment]
It is implemented in https://github.com/Shougo/unite.vim/pull/1004.
Closing.

**Comment by aquach** [comment]
FWIW, I've implemented a sorter that uses burke/matcher to do sorting instead of sorter_rank or sorter_selecta. His matcher is a clone of CommandT's fuzzy matching, which I've found to be smarter than selecta's and handles the case described in this thread and a bunch others that were frustrating me with selecta. My hacky code for it [lives here](https://gist.github.com/aquach/774b0a51e47301ff365d). If anyone has interest in it, feel free to tweak it and use it.
